### PR TITLE
Provide an clear overview of the asymmetric signature API

### PR DESCRIPTION
**Pull request by athoelke**
I've taken some ideas from #107 and #216 to clarify the documentation for asymmetric signatures, which seems valuable independently of those PRs:

* Rework the introduction to the signature chapter to describe:
  * Different signature schemes, and split the algorithm sub-chapter by scheme. This better paves the way for ML-DSA and SLH-DSA
  * The three types of algorithm: hash-and-sign, message-signature, and specialized-signature. This terminology is used throughout to improve clarity in how to use each algorithm.
  * The message and hash signature functions. This provides the right structure to add the interruptible APIs.
* Small improvements to the text in the functionality overview, to increase clarity.

### CODE REVIEW

File: doc/crypto/overview/functionality.rst [hunk 1]
```diff
@@ -114,15 +114,21 @@ Cryptographic operations
 
 For each type of cryptographic operation, the API can include:
 
-*   One or more *single-part* functions, which perform a whole operation in a single function call. For example, compute, verify, encrypt or decrypt.
-*   A *multi-part operation* --- which is a series of functions that work with a stored operation state.
+*   One or more *single-part* functions, which perform a whole operation in a single function call. For example, compute, verify, encrypt or decrypt. See :secref:`single-part-functions`.
+
+*   A *multi-part operation* --- which is a set of functions that work with a stored operation state. This provides more control over operation configuration, piecewise processing of large input data, or handling for multi-step processes. See :secref:`multi-part-operations`.
```
Comment (gilles-peskine-arm) [inline doc/crypto/overview/functionality.rst:119]: Preexisting: it's weird that “one or more single-part functions” is put in parallel with “a multi-part operation”. Each bullet point makes sense on its own, they're just weird when contrasted. Here's my attempt at polishing this.

> The API supports cryptographic operations through two kinds of interfaces:
>
> * A *single-part* function performs a whole operation in a single function call. For example, compute, verify, encrypt or decrypt. See :secref:`single-part-functions`.
> 
> *   A *multi-part operation* is a set of functions that work with a stored operation state. This provides more control over operation configuration, piecewise processing of large input data, or handling for multi-step processes. See :secref:`multi-part-operations`.
>
> Depending on the mechanism, one or both kind of interfaces may be provided.

**Review by gilles-peskine-arm — APPROVED** [review]
Looks good to me, with just some minor suggestions.

File: doc/crypto/overview/functionality.rst (hunk 1 above)
Comment (athoelke) [reply doc/crypto/overview/functionality.rst:119]: I like your proposed modification, and will adopt it.

File: doc/crypto/api/ops/signature.rst [hunk 2]
```diff
@@ -320,11 +507,11 @@ Asymmetric signature algorithms
...
 
     .. summary::
-        Edwards-curve digital signature algorithm with prehashing (HashEdDSA), using the Edwards448 curve.
+        Edwards-curve digital signature algorithm with pre-hashing (HashEdDSA), using the Edwards448 curve.
 
-    This algorithm can be used with both the message and hash signature functions.
+    This hash-and-sign signature algorithm can be used with both the message and hash signature functions.
 
-    This computes the Ed448ph algorithm as specified in :RFC-title:`8032#5.2`, and requires an Edwards448 curve key. An empty string is used as the context. The prehash function is the first 64 bytes of the output from SHAKE256.
+    This calculates the Ed448ph algorithm as specified in :RFC-title:`8032#5.2`, and requires an Edwards448 curve key. An empty string is used as the context. The pre-hash function is the first 64 bytes of the output from SHAKE256.
```
[the remark this answers is not in the document]
Comment (athoelke) [reply doc/crypto/api/ops/signature.rst]: Good idea - I've added cross references

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Pull request merged**